### PR TITLE
Prevent reduction of sources in "externally built" projects

### PR DESCRIPTION
**Pull request by setton**
This tests that subprojects declared as externally built do not get processed by adareducer.

### CODE REVIEW

File: ada_reducer/project_support.py [hunk 1]
```diff
@@ -31,3 +36,10 @@ def find(self, basename):
             print(f"file {basename} not found in project")
             return None
         return self.files[basename]
+
+    def belongs_to_project(self, filename):
+        """Return True iff filename is a source of this project tree,
+        not including externally built projects.
+        """
+        basename = PurePath(filename).name
+        return basename in self.files
```
Comment (glacambre) [inline ada_reducer/project_support.py:45]: Probably a stupid question but why aren't there any collisions in self.files? Like if I have `src/a/hello.adb` and `src/b/hello.adb`. Or if I have `hello.ads` and then `/wave/x86_64_linux/[...]/hello.ads`.

**Review by glacambre — APPROVED** [review]
Seems to work great on my big uwrap reproducer!

File: ada_reducer/project_support.py (hunk 1 above)
Comment (setton) [reply ada_reducer/project_support.py:45]: it's not supported in non-aggregate .gpr trees: one constraint is that each base file name should be unique.

Comment (glacambre) [reply ada_reducer/project_support.py:45]: I see. In any case, thank you for implementing this!

**Pull request merged**